opaque_keys/edx/keys.py

```diff
@@ -77,6 +84,37 @@ def block_type(self):  # pragma: no cover
         raise NotImplementedError()
 
 
+class CourseKeyV2(OpaqueKey):
+    """
+    An :class:`opaque_keys.OpaqueKey` identifying a
+    serialized Course Key object.
+    """
+    KEY_TYPE = 'course_key_v2'
+    __slots__ = ()
+
+    @abstractproperty
+    def from_course_run_key(self, course_key):  # pragma: no cover
+        """
+        Get course key v2 from the course run key.
+
+        Arguments:
+            course_key (:class:`CourseKey`): The course identifier.
+
+        """
+        raise NotImplementedError()
+
+    @abstractproperty
+    def make_course_run_key(self, course_run):  # pragma: no cover
```

Comment:
These were my suggestion. The existing keys typically are related by giving the more general key a way to create more specific keys by hydrating additional information.